File: python/binary_test.py
#!/usr/bin/python3

"""
Tests the binarytree class.
"""

import random, binarytree

list = []
for i in range(1000):
    list.append((i, random.random()))
random.shuffle(list)

tree = binarytree.Node(500, "MARKER")
for item in list:
    tree.insert(item[0], item[1])

print("Left:", len(tree._get_left([])))
print("Right:", len(tree._get_right([])))

print(tree.lookup(493))
print(tree.remove(493))

File: c/binarytest.c
/* This tests the binary search tree class */
#include <stdio.h>
#include "binarytree.h"

int main(int argc, const char *argv[])
{
    char action;
    while (action != 'e') {
        switch (action) {
            case 'n':
                printf("new node\n");
                printf("(n)ew, (l)ookup, (e)xit: ");
                break;
            case 'l':
                printf("lookup\n");
                printf("(n)ew, (l)ookup, (e)xit: ");
                break;
            default:
                printf("(n)ew, (l)ookup, (e)xit: ");
        }
        scanf(" %c", &action);
    }
    return 0;
}

